src/main/java/agh/ics/oop/SimulationEngine.java

```diff
@@ -24,6 +24,8 @@ private void addAnimalToMap(){
             Animal animal = new Animal(this.map, position);
             if (map.place(animal)){
                 animals.add(animal);
+                //////////////////////////////////////////////////////////////////
+                animal.addObserver((IPositionChangeObserver) this.map);
```

Comment:
Jeśli każda mapa rozszerzałaby ten interfejs to nie musielibyśmy rzutować -> interfejsy też mogą rozszerzać inne interfejsy.